keps/sig-architecture/1618-conformance-profiles/README.md

```diff
@@ -0,0 +1,444 @@
+minimum is capable of running stateless user workloads. From this user’s point
+of view, cluster administration functions are not required.
+
+Profiles also support a simplified model for determining if a given application
+that has been packaged to run on Kubernetes (e.g., in a Helm chart) is able to
+run on a given cluster. If clusters are classified into a small (~5-10) number
+of profiles that define broad functional boundaries, applications will be able
+to document the profiles required in order for them to run.
+
+### Goals
+
+ * Design and implement a mechanism to express profiles in the conformance
+   behavior definitions.
+ * Design and implement a mechanism to validate whether a cluster conforms to
+   each profile.
+ * Document guidance on how to choose the scope of individual profiles.
+ * Document guidance on how to decide the profile in which a new feature
+   belongs.
+ * Define a base profile and at least one other profile.
+
+### Non-Goals
+
+ * Define a complete set of profiles that covers all behaviors. This is an
+   ongoing effort for the conformance subproject.
+ * Complete all tests to validate all profiles. This is an ongoing effort for
+   the conformance subproject.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Role: Developer
+
+As a developer of an optional feature, I must be able to define behaviors for
+that feature that a vendor claiming support for the feature must follow.
+
+#### Role: Plugin API & Plugin Developers
+
+As a developer a pluggable API for Kubernetes, such as CNI, CSI, CRI, and
+Ingress, I must be able to define behaviors which I expect plugins to follow,
+and verify that a specific plugin properly implements those behaviors.
+
+As an developer of a plugin for Kubernetes, I must be able to verify that my
+plugin conforms to the expected behaviors.
+
+#### Role: Application Packager
+
+As an application packager preparing an application for deployment on
+Kubernetes, I must be able to identify the profiles required to run the packaged
+workloads.
+
+#### Role: End User
+
+As an end user deploying an application on Kubernetes, I must be able to
+identify the profiles supported by my cluster and the application I wish to
+deploy.
+
+ - Should be able to define which tests to run up-front (eg: conformance, plus
+   profiles, profiles only, etc)
+ - May be able to detect or enumerate list of eligible profiles for a given
```

Comment:
is this via some query-able endpoint under discussion?

Reply:
The idea of polling for features would be great.

Reply:
We haven't ironed out the technical details yet, but this is a great suggestion.